### PR TITLE
[v6r19] Fix StorageElement inheritance and add a new Resources section

### PR DESCRIPTION
**Pull request by aldbr**
This pull request:

+ fixes different weird behaviours in Storage Element, as the inheritance of the protocol sections
+ creates a new Resources section named *StorageElementBases* containing only Base SEs
+ updates the documentation

### CODE REVIEW

**Comment by aldbr** [comment]
I just add a commit for a very little condition improvement in StorageFactory

**Comment by chaen** [comment]
providing the tests succeed, review OK.
Please add in the wiki for v6r19 that as of this release, ```ProtocolName``` is not supported anymore in the configuration and must be replaced by ```PluginName```

**Comment by aldbr** [comment]
Alright, I added the information

File: docs/source/AdministratorGuide/Systems/DataManagement/index.rst [hunk 1]
```diff
@@ -43,92 +43,125 @@ DIRAC provides an abstraction of a SE interface that allows to access different
...
         Access = remote
-        SpaceToken = LHCb_USER
+        SpaceToken = LHCb-EOS
         WSUrl = /srm/v2/server?SFN=
       }
-   }
+    }
 
-Note that in case of overwriting, the protocol sections are matched using the PluginName attribute, and not the section name, which is irrelevant at all.
+Note that base SE must be separated from the inherited SE in two different sections. You can also notice that the name of the protocol section can be a plugin name. In this way, you do not need to specify a plugin name inside.
```
Comment (phicharp) [inline docs/source/AdministratorGuide/Systems/DataManagement/index.rst]: Don't we want to make that a rule at some point?
The current implementation eases the migration, but at some point we should not have the PluginName any longer, right?

**Comment by chaen** [comment]
We must have it if we ever want to use the same plugin twice (and we will want soon I suspect ;-) )

**Comment by chaen** [comment]
For reference, this was agreed at this bild dev: https://indico.cern.ch/event/608327/

File: docs/source/AdministratorGuide/Systems/DataManagement/index.rst (hunk 1 above)
Comment (atsareg) [inline docs/source/AdministratorGuide/Systems/DataManagement/index.rst]: My experience shows that giving "meaningful" names to sections is often confusing. So, please, keep PluginName as a mandatory parameter and do not try to interpret the section name. 

Eventually (not now), I suggest to introduce yet another section, e.g. StoragePlugins, in the SE definition in which all the per plugin sections will go. Then there will be no confusion that each section corresponds to a plugin. This is how it is done for Queues in the CE sections. So, SEs should be described symmetrically. But even in this case I would still keep PluginName as mandatory. This is what was foreseen for v7r0. A crude example below:  

  StorageElements
  {
    SE1
    {
       Option1 = value
       Option2 = value
       StoragePlugins 
       {
          SomeName_1
          {
              PluginName = name1
              ...
           } 
           SomeName_2
           {
               PluginName = name 2
            }
       }
    }     
  }

**Comment by atsareg** [comment]
We now have a CS section inheritance in several places. We should think of making it a generic feature of the CS. So, that the inheritance is only interpreted by the CS client and all the other consumers of the CS data are just getting a ready to use data structure

**Comment by phicharp** [comment]
AFAIS, what was implemented is exactly what had been approved in February. Backward compatibility has been preserved as requested.
Therefore I don't see why this PR was closed. If users want to keep the previous behavior, they can keep it... We in LHCb will use the new feature which is much clearer than having sections called randomly `AccessProtocol.1` and for which one doesn't know what they mean... 
@atsareg: please re-open this PR

**Comment by atsareg** [comment]
This was not at all my intention to close the PR. Just pressed the wrong button when commenting. Sorry about that.

**Comment by phicharp** [comment]
Why the review is limited to the few developers who have write access to the repository? This is absurd, isn't it?

**Comment by phicharp** [comment]
@aldbr : Alexandre, you should rebase your PR with integration apparently...

**Comment by aldbr** [comment]
@phicharp Ok, I rebased it with integration

**Comment by atsareg** [comment]
@phicharp: review is indeed limited by the Github to those who have write privileges for the repo. But we know each other. So, if you leave a comment "Review OK", this is taken into account in the same way as other reviews when considering merging a given PR.

**Comment by phicharp** [comment]
Review OK (again)

**Pull request merged**